ephios/core/notifications/backends.py

```diff
@@ -0,0 +1,84 @@
+from django.core.mail import EmailMultiAlternatives
+from django.utils.translation import gettext_lazy as _
+from webpush import send_user_notification
+
+from ephios.core.models.users import Notification
+from ephios.core.notifications.types import notification_type_from_slug
+from ephios.core.signals import register_notification_backends
+
+
+def installed_notification_backends():
+    for _, backends in register_notification_backends.send_to_all_plugins(None):
+        yield from (b() for b in backends)
+
+
+def enabled_notification_backends():
+    for _, backends in register_notification_backends.send(None):
+        yield from (b() for b in backends)
+
+
+class AbstractBackend:
+    @property
+    def slug(self):
+        return NotImplementedError
+
+    @property
+    def title(self):
+        return NotImplementedError
+
+    @classmethod
+    def can_send(cls, notification):
+        return notification.user is not None
+
+    @classmethod
+    def user_prefers_sending(cls, notification):
+        notification_type = notification_type_from_slug(notification.slug)
+        if notification_type.unsubscribe_allowed and notification.user is not None:
+            backends = notification.user.preferences["notifications__notifications"].get(
+                notification_type.slug
+            )
+            if backends is not None:
+                return cls.slug in backends
+        return True
+
+    @classmethod
+    def send(cls, notification: Notification):
+        raise NotImplementedError
+
+
+class EmailBackend(AbstractBackend):
+    slug = "ephios_backend_email"
+    title = _("via email")
+
+    @classmethod
+    def can_send(cls, notification):
+        return notification.user is not None or "email" in notification.data
+
+    @classmethod
+    def _get_mailaddress(cls, notification):
+        return notification.user.email if notification.user else notification.data.get("email")
+
+    @classmethod
+    def send(cls, notification):
+        notification_type = notification_type_from_slug(notification.slug)
+        email = EmailMultiAlternatives(
+            to=[cls._get_mailaddress(notification)],
+            subject=notification_type.get_subject(notification),
+            body=notification_type.as_plaintext(notification),
+        )
+        email.attach_alternative(notification_type.as_html(notification), "text/html")
```

Comment:
I was asking myself, why `as_html` isn't a method on `Notification`?
Couldn't you just fetch the notification type and render the contents there instead of having to do this in the backend?
It's one more call, but it seems more clean to me.